cmd/homi/setup/cmd.go

```diff
@@ -1107,13 +1105,17 @@ func indexGenType(genTypeFlag string, base string) int {
 func findGenType(ctx *cli.Context) int {
 	var genType int
 	if ctx.Args().Present() {
-		genType = indexGenType(ctx.Args()[0], "")
+		genType = indexGenType(ctx.Args().First(), "")
```

Comment:
It's quite confusing to me. What about next?
```
	var (
		genTypeName string
		baseString string
		genType int
	)

	if ctx.Args().Present() {
		genTypeName, baseString = ctx.Args().First(), ""
	} else {
		genTypeName, baseString = ctx.String(genTypeFlag.Name), Types[0]
	}

	genType = indexGenType(genTypeName, baseString)
	if genType == TypeNotDefined {
		fmt.Printf("Wrong Type : %s\nSupported Types : [docker, local, remote, deploy]\n\n", genTypeName)
		cli.ShowSubcommandHelp(ctx)
		os.Exit(1)
	}
```

Reply:
Resolved at 5afbed2c5c92c74839b9b4df1027bf423f339a3d